runtime/onert/api/src/nnfw_api_internal.cc

```diff
@@ -1450,8 +1454,64 @@ NNFW_STATUS nnfw_session::train_export_circle(const char *path)
     return NNFW_STATUS_INVALID_STATE;
   }
 
-  // NYI
-  return NNFW_STATUS_ERROR;
+  class MMappedFile
+  {
+  public:
+    MMappedFile(const char *filename) { _fd = open(filename, O_RDWR); }
+
+    bool ensure_mmap()
+    {
+      struct stat file_stat;
+      if (fstat(_fd, &file_stat) != 0)
+        return false;
+      // file_stat.st_size is `off_t` while msync, munmap requires `size_t`
+      // `off_t` may be long or long long.
+      // `size_t` may be uint32 or uint64.
+      assert(sizeof(off_t) == sizeof(size_t));
```

Comment:
```suggestion
      static_assert(sizeof(off_t) == sizeof(size_t));
```

Reply:
@ragmani [Reference](https://en.cppreference.com/w/cpp/language/static_assert) says `static_assert( bool-constexpr )` is available since c++17. It may require additional change. Let's see how our CI says.

Reply:
If so, how about using `static_assert( bool-constexpr , message )` ?